master/internal/job/jobservice/jobservice.go

```diff
@@ -206,19 +206,9 @@ func (s *Service) applyUpdate(update *jobv1.QueueControl) error {
 		}
 		return j.SetResourcePool(action.ResourcePool)
 	case *jobv1.QueueControl_AheadOf:
-		return s.rm.MoveJob(sproto.MoveJob{
-			ID:           jobID,
-			Anchor:       model.JobID(action.AheadOf),
-			Ahead:        true,
-			ResourcePool: j.ResourcePool(),
-		})
+		return fmt.Errorf("action not supported - update priority to move job")
```

Comment:
we should update the proto definition to no longer have these options

Reply:
If we leave it like this, we can have a more informative error message though right?

I was thinking we could leave it until the UI changes are done to actually remove the option.

Reply:
The UI updates need to be prioritized. That could be in a future release. My understanding from product is that it was an acceptable trade-off.

https://hpe-aiatscale.slack.com/archives/C06GMG83ZE0/p1718299032071299?thread_ts=1718298903.798219&cid=C06GMG83ZE0

I agree with marking the proto definition as deprecated.

Reply:
Shouldn't the UI happen removal happen first? 

Can we just wait till the UI removal happens then?

Reply:
I don't think I understand what you're suggesting. Are you suggesting we shouldn't merge this PR until the corresponding front-end work is completed?

My understanding from previous conversations is that we were going to remove as much as possible from the backend, and leave it as a backlog task to update the front-end. https://hpe-aiatscale.slack.com/archives/C06GMG83ZE0/p1718311967982659?thread_ts=1718298903.798219&cid=C06GMG83ZE0

Reply:
That previous conversation is talking about a slightly different situation. The previous conversation assumed that Kubernetes would still support this feature and the backlog ticket is to grey out it for agentrm / slurm cases. This change now has removed the feature from all resource managers. I think there is a gap between having a button that only works in certain situations verse a button that never works.

Have we asked the frontend team if by the release would be a reasonable deadline for their work on this?